src/components/modebar/buttons.js

```diff
@@ -0,0 +1,495 @@
+/**
+* Copyright 2012-2015, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Plotly = require('../../plotly');
+var Lib = require('../../lib');
+var Snapshot = require('../../snapshot');
+
+
+var modebarButtons = module.exports = {};
```

Comment:
i.e the Modebar constructor expects the button config objects as input, but doesn't know anything about them.